libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -681,12 +753,22 @@ export class PackageExtractor {
           //
           // We can detect the duplicates by comparing the path.resolve() result.
           const copySourcePath: string = path.resolve(sourceFolderPath, npmPackFile);
+
+          // Filter out files that are excluded by the project configuration or dependency configuration.
+          if (
+            (isLocalProject && isFileExcludedForProject(npmPackFile)) ||
+            (!isLocalProject && isFileExcludedForDependency(npmPackFile))
+          ) {
+            return;
+          }
```

Comment:
Move this before the `path.resolve`. It's more expensive than it has any right to be, so we should avoid doing it if we can.